mongo/src/org/immutables/mongo/bson4gson/GsonCodecs.java

```diff
@@ -150,9 +152,35 @@ public <T> Codec<T> get(Class<T> clazz) {
     };
   }
 
+  /**
+   * Appends a TypeAdapterFactory to allow Gson and Bson adapters to co-exist.
+   * Avoids registering the TypeAdapterFactory if Gson already has factories for BSON types.
+   *
+   * @param gson preconfigured instance
+   * @return {@code gson} configured with delegating type adapter factory.
+   */
+  public static Gson newGsonWithDelegatingTypeAdapterFactory(Gson gson) {
```

Comment:
Maybe call it `newGsonWithMongoSupport` or `withBsonSupport(Gson)` or `addBsonSupport(Gson)`.

`WithDelegatingTypeAdapterFactory` doesn't fully convey the meaning of this method.

Reply:
right, it's more important what the purpose of the action is (adds bson support) is than in which way it's technically achieved (by delegating type adapter factory)

Reply:
addressed by renaming it to `newGsonWithBsonSupport`